lib/rubygems/uninstaller.rb

```diff
@@ -88,10 +88,18 @@ def uninstall
       list << spec
     end
 
+    if list.empty?
+      raise Gem::InstallError, "gem #{@gem.inspect} is not installed"
```

Comment:
Don't you think we could instead have an error class `Gem::UninstallError` when we do uninstall? 

It is probably a breaking change... :/

Reply:
Absolutely, it was very confusing the first time I saw this. I'll try to change this in the future.